internal/utils/logger/logger.go

```diff
@@ -0,0 +1,60 @@
+package logger
+
+import (
+	"io"
+	"log/slog"
+	"os"
+	"sync"
+
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+type LoggerConfig struct {
+	FilePath         string
+	UserLocalTime    bool
+	FileMaxSizeInMB  int
+	FileMaxBackups   int
+	FileMaxAgeInDays int
+	FileCompress     bool
+	LogLevel         slog.Level
+}
+
+var once sync.Once
+
+var logger *slog.Logger
+
+var loggerConfig LoggerConfig
+
+func init() {
+
+	loggerConfig = LoggerConfig{
+		FilePath:         "logs/logs.log",
+		UserLocalTime:    false,
+		FileMaxSizeInMB:  10,
+		FileMaxBackups:   3,
+		FileMaxAgeInDays: 30,
+		FileCompress:     false,
+		LogLevel:         slog.LevelDebug,
+	}
+}
+
+func Get() *slog.Logger {
+
+	once.Do(func() {
```

Comment:
TIL about `once` - is this considered the idiomatic was to create a singleton in go?

Reply:
Well, I figured this way would mean less code changes vs a dependency injected logger?

Reply:
No objections to your approach, just asking since I hadn't seen this before. No need for dependency injection since i don't think there will ever be a need to inject a different logger for testing